src/api/schema/filter.rs

```diff
@@ -46,8 +47,38 @@ impl StringFilter {
 }
 
 #[derive(InputObject)]
-#[graphql(concrete(name = "ComponentKindFilter", params(ComponentKind)))]
 #[graphql(concrete(name = "SourceOutputTypeFilter", params(source::SourceOutputType)))]
+// Filter for GraphQL lists
+pub struct ListFilter<T: InputType + PartialEq + Eq + Ord> {
+    pub equals: Option<Vec<T>>,
+    pub not_equals: Option<Vec<T>>,
+    pub contains: Option<T>,
+    pub not_contains: Option<T>,
+}
+
+impl<T: InputType + PartialEq + Eq + std::cmp::Ord> ListFilter<T> {
```

Comment:
Nit- `Ord` is already in scope:

```suggestion
impl<T: InputType + PartialEq + Eq + Ord> ListFilter<T> {
```